adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/timeutils/TimeUtilsImpl.java

```diff
@@ -81,43 +84,36 @@ public Integer convertHHMMSSToIntFromNoonOfDayOfService(final String timeAsStrin
+    public String convertIntegerToHMMSS(final Integer elapsedDurationSinceNoonInSeconds) {
+        if (elapsedDurationSinceNoonInSeconds != null) {
+            final LocalTime noon = LocalTime.NOON;
+            // elapsedDurationSinceNoonInSeconds > 0 for times after noon, e.g 14:00PM is represented a +2*3600s=+7200s
+            // elapsedDurationSinceNoonInSeconds < 0 for times before noon, e.g 11:00AM is represented as -1*3600s=-3600s
+            final boolean isTimeGreaterThan24hours =
+                    elapsedDurationSinceNoonInSeconds >= TimeUnit.HOURS.toSeconds(12);
+
+            if (!isTimeGreaterThan24hours) {
+                return String.format("%02d:%02d:%02d", noon.plusSeconds(elapsedDurationSinceNoonInSeconds).getHour(),
+                        noon.plusSeconds(elapsedDurationSinceNoonInSeconds).getMinute(),
+                        noon.plusSeconds(elapsedDurationSinceNoonInSeconds).getSecond());
             } else {
-                // Determine number of minutes elapsed minus the number of hours elapsed
-                minuteValue = ((int) (TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon) -
-                        TimeUnit.HOURS.toMinutes(TimeUnit.SECONDS.toHours(elapsedDurationSinceNoon)))) % 60;
-                if (minuteValue < 0) {
-                    hourValue = hourValue - 1;
-                    minuteValue = 1 + ((int) -(TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon) -
-                            TimeUnit.HOURS.toMinutes(TimeUnit.SECONDS.toHours(elapsedDurationSinceNoon))));
-                }
-                // Determine number of seconds elapsed minus the number of hours and minutes elapsed
-                secondValue = (int) (TimeUnit.SECONDS.toSeconds(elapsedDurationSinceNoon) -
-                        TimeUnit.MINUTES.toSeconds(TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon)));
-                if ((int) (TimeUnit.SECONDS.toSeconds(elapsedDurationSinceNoon) -
-                        TimeUnit.MINUTES.toSeconds(TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon))) < 0) {
-                    secondValue = 60 + secondValue;
-                }
+                // This wraps around midnight
+                LocalTime timeAfterMidnight = noon.plusSeconds(elapsedDurationSinceNoonInSeconds);
+                // Get the number of hours, minutes, and seconds between midnight and timeAfterMidnight, which is the
+                // time to add to 24:00:00
+                long hoursAfterMidnight = LocalTime.MIDNIGHT.until(timeAfterMidnight, ChronoUnit.HOURS);
+                long minutesAfterMidnight = LocalTime.MIDNIGHT.until(timeAfterMidnight, ChronoUnit.MINUTES) % 60;
+                long secondsAfterMidnight = LocalTime.MIDNIGHT.until(timeAfterMidnight, ChronoUnit.SECONDS) % 60;
+                return String.format("%02d:%02d:%02d",
+                        hoursAfterMidnight + 24,
```

Comment:
In retrospect, this is probably clearer as:

```suggestion
                        24 + hoursAfterMidnight,
```